### DISCUSSION TITLE
VerticalStackLayout Spacing with hidden children

### DISCUSSION THREAD

**Discussion by johnl188** [dotnet/maui#9585] [Q&A] [answered]
I've been messing around with VerticalStackLayout where some of the children are conditionally displayed. When a child is hidden, the Spacing attribute of the VerticalStackLayout is still added to the size of the layout so there appears to be extra space at the bottom of the VerticalStackLayout. I was hoping hiding a child would also hide the spacing associated with that child.

Looking at the source code for the VerticalStackLayoutManager, in the Size Method it makes sense why the spacing is still added.
`measuredHeight += MeasureSpacing(Stack.Spacing, Stack.Count);` is used to add spacing for each child in the stack. I created my own custom layout to only add spacing for the visible children count and that results in the layout that I expected. 

I was wondering if anyone has encountered this before. A solution that I originally found was converting to a grid with a bunch of auto sized rows, but I didn't like that and wanted to make the VerticalStackLayout work.

Would it make sense to update VerticalStackLayout to only add spacing for visible rows instead of all rows?

**Comment by hartez** [comment, marked as answer]
@johnl188 Yep, this is a bug in the VSL implementation (and also in the HorizontalStackLayout). `MeasureSpacing` should _not_ be adding spacing for invisible items. I'm going to convert this to a GitHub issue so we can track it and fix it.

**Reply by johnl188** [reply]
Awesome, thank you.

**Reply by jmichas** [reply]
So, in the latest dotnet 9.0.100 this still appears to be a problem. I had some conditional elements in a vertical stack and when IsVisible is false they take up space. Confirmed by commenting out the elements and the layout spacing looks as expected.

The issue is locked so I figured I'd comment here, hopefully this can get fixed.

**Reply by jmichas** [reply]
Ok, sorry, never mind. Turns out there was a custom control with ContentView that had a stack layout in it that was visible, just empty. That takes up space and causes the vertical stack to calulate padding.

**Comment by last-Programmer** [comment]
I have this problme with maui 10 also. when i set the isvisible of child to true or false the parent coantiner grid is not resized. it seems to be that i need to call the invalidatemeasure in thses cases. This is not being done automaticcally.
